packages/core/src/variables/VariableProcessor.js

```diff
@@ -396,8 +415,9 @@ class VariablePreprocessor {
    * @param content to render
    * @param highestPriorityVariables to render with the highest priority if any.
    *        This is currently only used for the MAIN_CONTENT_BODY in layouts.
+   * @param keepPercentRaw whether to reoutput {% raw/endraw %} tags, also used only for layouts.
```

Comment:
Since we have some specific behaviour for layouts, do you think it's worth having a `renderLayout` function?

Reply:
that makes sense! I think there's quite a bit of opportunity for refactor regarding layouts though, apart from this. (site nav, page layouts, header, footer takes up a sizeable portion of `Page.js`) Perhaps we could defer it to that pr?

Reply:
That sounds fine too, let's fix that in the next PR then 🚀